holmes/plugins/skills/builtin/kubernetes-troubleshooting/SKILL.md

```
@@ -0,0 +1,46 @@
+## Workflow
+
+* run as many kubectl commands as you need to gather more information, then respond.
+* if possible, do so repeatedly on different Kubernetes objects.
+* for example, for deployments first run kubectl on the deployment then a replicaset inside it, then a pod inside that.
+* when investigating a pod that crashed or application errors, always run kubectl_describe and fetch the logs
+* Do check both the status of the kubernetes resources and the application runtime as well, by investigating logs
+* do not give an answer like "The pod is pending" as that doesn't state why the pod is pending and how to fix it.
+* do not give an answer like "Pod's node affinity/selector doesn't match any available nodes" because that doesn't include data on WHICH label doesn't match
+* if investigating an issue on many pods, there is no need to check more than 3 individual pods in the same deployment. pick up to a representative 3 from each deployment if relevant
+* if the user says something isn't working, ALWAYS:
+** use kubectl_describe on the owner workload + individual pods and look for any transient issues they might have been referring to
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)